slack/commands/rate.go

```diff
@@ -0,0 +1,71 @@
+package commands
+
+import (
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+
+	"github.com/sjuls/soup-ranking/dbctx"
+)
+
+type (
+	rateFlags struct {
+		Score   int
+		Comment string
+	}
+
+	rateCommand struct {
+		repo dbctx.ScoreRepository
+	}
+)
+
+// NewRateCommand create a new rate command
+func NewRateCommand(repo dbctx.ScoreRepository) Command {
+	return &rateCommand{repo}
+}
+
+func (c *rateCommand) Execute(args string, output io.Writer) {
+	rateMatcher, err := regexp.Compile("([1-9]|10)\\s(.*)")
+	if err != nil {
+		fmt.Fprintln(output, err.Error())
+		return
+	}
+
+	rateMatches := rateMatcher.FindStringSubmatch(args)
+	scoreValue, err := strconv.ParseInt(rateMatches[1], 10, 32)
+
+	if err != nil {
+		fmt.Fprintln(output, err.Error())
+		return
+	}
+
+	if scoreValue < 1 || 10 < scoreValue {
+		fmt.Fprintln(output, "Score must be between 1 and 10")
+		return
+	}
+
+	score := dbctx.Score{
+		Score: int(scoreValue),
+	}
+
+	if len(args) > 2 {
+		score.Comment = rateMatches[2]
+	}
+
+	err = c.repo.SaveScore(&score)
+
```

Comment:
Also this one

Reply:
Why? Linebreaks are like full stops - gives the developer reading it time to breathe.

Reply:
I see your point of view